### DISCUSSION TITLE
Support for uninstalling a plugin

### DISCUSSION THREAD

**Discussion by alfredbuehler** [aljawaid/PluginManager#67] [Ideas]
Here is my idea: If the plugin developer added a function to the schema files, like
```php
function uninstall(PDO $pdo)
{
    $pdo->exec("ALTER TABLE subtasks DROP due_description");
}
```
Adding the function is voluntary but recommended. If the function is present, the PluginManager could clean up the database after unloading the plugin. The developer hopefully knows exactly, what to clean up, where tools like ContentCleaner only can guess.

Any thoughts?
abu

**Comment by aljawaid** [comment]
That seems like a very good idea if implemented properly but not many people are updating their plugins over the last few months except for me, you and @creecros so is it worth all the effort?

Also, if you put the above snippet in the schema file, how would you call it on uninstall?  I mean if we just add the support then we can adjust our own plugins I guess at least, others will ignore the function.

**Reply by alfredbuehler** [reply]
Of course, you're fully right, maybe it isn't worth any effort. It was just an idea that came up.
